### PR TITLE
[OSDOCS-5013]: CPMS for GCP: Supported features

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.13+

Issue:
[OCPCLOUD-1774](https://issues.redhat.com//browse/OCPCLOUD-1774) | [OSDOCS-5013](https://issues.redhat.com//browse/OSDOCS-5013)

Link to docs preview:
[Enabling Google Cloud Platform features for control plane machines](https://56305--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-using.html#cpmso-supported-features-gcp_cpmso-using)

QE review:
- [x] QE has approved this change.

Additional information:
` machineset-gcp-enabling-customer-managed-encryption.adoc` shows up as new because I renamed it. Changes are:
* Removed "compute" from "compute machine set" because it applies to control plane too
* Added `cpmso` condition
* Fixed YAML to show full required API info
* Minor wording tweaks

### CODE REVIEW

File: modules/machineset-gcp-pd-disk-types.adoc [hunk 1]
```diff
@@ -1,30 +1,50 @@
...
 
 . Edit the following line under the `providerSpec` field:
 +
 [source,yaml]
 ----
-providerSpec:
-  value:
-    disks:
-      type: <pd-disk-type> <1>
+apiVersion: machine.openshift.io/v1
```
Comment (JoelSpeed) [inline modules/machineset-gcp-pd-disk-types.adoc:26]: MachineSets are `machine.openshift.io/v1beta1` so need to move this inside the if statement

File: modules/machineset-gcp-enabling-customer-managed-encryption.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  --location <key_ring_location> \
+  --member "serviceAccount:service-<project_number>@compute-system.iam.gserviceaccount.com” \
+  --role roles/cloudkms.cryptoKeyEncrypterDecrypter
+----
+
+. Configure the encryption key under the `providerSpec` field in your machine set YAML file. For example:
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1
```
Comment (JoelSpeed) [inline modules/machineset-gcp-enabling-customer-managed-encryption.adoc:40]: MachineSets are `machine.openshift.io/v1beta1` so need to move this inside the if statement

File: modules/machineset-gcp-pd-disk-types.adoc (hunk 1 above)
Comment (jeana-redhat) [reply modules/machineset-gcp-pd-disk-types.adoc:26]: aaaaah very good, thanks!

**Comment by JoelSpeed** [comment]
LGTM

**Comment by jeana-redhat** [comment]
@miyadav this is ready for verification, PTAL

**Comment by jeana-redhat** [comment]
@miyadav this is now truly ready for verification, PTAL :)

**Comment by jeana-redhat** [comment]
@miyadav thanks, I will add a note with that information.

File: modules/machineset-gcp-shielded-vms.adoc [hunk 3]
```diff
@@ -45,6 +45,11 @@ spec:
 <1> In this section, specify any Shielded VM options that you want.
 <2> Specify whether UEFI Secure Boot is enabled. Valid values are `Disabled` or `Enabled`.
 <3> Specify whether integrity monitoring is enabled. Valid values are `Disabled` or `Enabled`.
++
+[NOTE]
+====
+When integrity monitoring is enabled, you must not disable the virtual trusted platform module (vTPM).
+====
 <4> Specify whether virtual trusted platform module (vTPM) is enabled. Valid values are `Disabled` or `Enabled`.
```
Comment (bscott-rh) [inline modules/machineset-gcp-shielded-vms.adoc:53]:
```suggestion
====
+
<4> Specify whether virtual trusted platform module (vTPM) is enabled. Valid values are `Disabled` or `Enabled`.
```
The 4th annotation is showing up as `<1>`. I think it's missing a `+` before it.

**Review by bscott-rh — APPROVED** [review]
Couple of minor things, otherwise LGTM.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/machineset-gcp-shielded-vms.adoc (hunk 3 above)
Comment (jeana-redhat) [reply modules/machineset-gcp-shielded-vms.adoc]: Both great catches. Will wait for this fix to build and verify the output. TYSM!

**Pull request merged**